nucypher/cli/protocol.py

```diff
@@ -0,0 +1,153 @@
+"""
+This file is part of nucypher.
+
+nucypher is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+nucypher is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with nucypher.  If not, see <https://www.gnu.org/licenses/>.
+"""
+
+
+import os
+from collections import deque
+
+import click
+import maya
+from twisted.internet import reactor
+from twisted.protocols.basic import LineReceiver
+
+from nucypher.config.constants import SEEDNODES
+
+
+class UrsulaCommandProtocol(LineReceiver):
+
+    encoding = 'utf-8'
+    delimiter = os.linesep.encode(encoding=encoding)
+
+    def __init__(self, ursula):
+        self.ursula = ursula
+        self.start_time = maya.now()
+        self.prompt = bytes('Ursula({}) >>> '.format(self.ursula.checksum_public_address[:9]), encoding='utf-8')
+        super().__init__()
+
+    def _paint_known_nodes(self):
+        # Gather Data
+        known_nodes = self.ursula.known_nodes
+        known_certificate_files = os.listdir(self.ursula.known_certificates_dir)
+        number_of_known_nodes = len(known_nodes)
+        seen_nodes = len(known_certificate_files)
+
+        # Operating Mode
+        federated_only = self.ursula.federated_only
+        if federated_only:
+            click.secho("Configured in Federated Only mode", fg='green')
+
+        # Heading
+        label = "Known Nodes (connected {} / seen {})".format(number_of_known_nodes, seen_nodes)
+        heading = '\n' + label + " " * (45 - len(label)) + "Last Seen    "
+        click.secho(heading, bold=True, nl=False)
+
+        # Legend
+        color_index = {
+            'self': 'yellow',
+            'known': 'white',
+            'seednode': 'blue'
+        }
+        for node_type, color in color_index.items():
+            click.secho('{0:<6} | '.format(node_type), fg=color, nl=False)
+        click.echo('\n')
+
+        seednode_addresses = list(bn.checksum_address for bn in SEEDNODES)
+        for address, node in known_nodes.items():
```

Comment:
This is broken since `FleetStateTracker`. Suggested change:
```suggestion
        for node in known_nodes:
```

Reply:
:heavy_check_mark: